app/models/staff_response.rb

```diff
@@ -5,6 +5,8 @@ class StaffResponse
   attr_accessor :visit, :user
 
   before_validation :check_slot_available
+  before_validation :check_lead_visitor_not_banned
+  before_validation :check_lead_visitor_on_list
 
   validate :validate_visit_is_processable
   validate :visit_or_rejection_validity
```

Comment:
I'd rather change this in a separate PR as part of the extra validation/scenarios needed to book to nomis